src/__tests__/medications/requests/NewMedicationRequest.test.tsx

```diff
@@ -1,186 +1,205 @@
-      medication: 'expected medication',
-      status: 'draft',
-      notes: 'expected notes',
-      id: '1234',
-      requestedOn: expectedDate.toISOString(),
-    } as Medication
-    const store = mockStore({
-      medication: { status: 'loading', error: {} },
-      user: { user: { id: 'fake id' } },
-    } as any)
-    beforeEach(() => {
-      jest.resetAllMocks()
-      Date.now = jest.fn(() => expectedDate.valueOf())
-      medicationRepositorySaveSpy = jest
-        .spyOn(MedicationRepository, 'save')
-        .mockResolvedValue(expectedMedication as Medication)
-
-      jest
-        .spyOn(PatientRepository, 'search')
-        .mockResolvedValue([
-          { id: expectedMedication.patient, fullName: 'some full name' },
-        ] as Patient[])
-    })
-
     it('should save the medication request and navigate to "/medications/:id"', async () => {
-      const { wrapper, history } = await setup(store)
-
-      const patientTypeahead = wrapper.find(Typeahead)
-      await act(async () => {
-        const onChange = patientTypeahead.prop('onChange')
-        await onChange([{ id: expectedMedication.patient }] as Patient[])
+      const { history } = setup()
+      const patient = screen.getByPlaceholderText(/medications\.medication\.patient/i)
+      const medication = screen.getByPlaceholderText(/medications\.medication\.medication/i)
+      const medicationNotes = screen.getByRole('textbox', {
+        name: /medications\.medication\.notes/i,
       })
-
-      const medicationInput = wrapper.find(TextInputWithLabelFormGroup).at(0)
-      act(() => {
-        const onChange = medicationInput.prop('onChange') as any
-        onChange({ currentTarget: { value: expectedMedication.medication } })
-      })
-
-      const notesTextField = wrapper.find(TextFieldWithLabelFormGroup)
-      act(() => {
-        const onChange = notesTextField.prop('onChange') as any
-        onChange({ currentTarget: { value: expectedMedication.notes } })
-      })
-      wrapper.update()
-
-      const saveButton = wrapper.find(Button).at(0)
-      await act(async () => {
-        const onClick = saveButton.prop('onClick') as any
-        await onClick()
-      })
-
-      expect(medicationRepositorySaveSpy).toHaveBeenCalledTimes(1)
-      expect(medicationRepositorySaveSpy).toHaveBeenCalledWith(
-        expect.objectContaining({
-          patient: expectedMedication.patient,
-          medication: expectedMedication.medication,
-          notes: expectedMedication.notes,
-          status: 'draft',
-          requestedOn: expectedDate.toISOString(),
+      const medStatus = screen.getAllByPlaceholderText('-- Choose --')[0]
+      const medicationIntent = screen.getAllByPlaceholderText('-- Choose --')[1]
+      const medicationPriority = screen.getAllByPlaceholderText('-- Choose --')[2]
+
+      userEvent.type(patient, 'Bruce Wayne')
+      userEvent.type(medication, 'Ibuprofen')
+      userEvent.type(medicationNotes, 'Be warned he is Batman')
```

Comment:
LOL, that he is

Reply:
@emma-r-slight I think this was you lol I am a huge fan of this kind of mock data! 😆